R/markdown.R

```diff
@@ -87,5 +87,7 @@ markdown_text2 <- function(text, pkg, ...) {
     return(paste0(xml2::xml_contents(children), collapse=""))
   }
 
-  return(paste0(as.character(children), collapse=""))
+  output <- paste0(as.character(children), collapse="")
```

Comment:
The output of this function feels a bit murky to me — sometimes it returns inline elements and some times it returns a block element. That means depending on what the user supplies, we might generate invalid HTML. Or in other words — we need to decide whether a markdown elements is inline or block, not the user, so I think it should be an argument to the function. If the user supplies blocks where we expect inline (or vice versa) we need to generate a clear error message.